shark_turbine/kernel/wave/constraints.py

```diff
@@ -46,6 +45,7 @@ class HardwareConstraint(Constraint):
     threads_per_wave: int
     waves_per_block: Optional[tuple[int, int, int]] = None
     mma_type: Optional[MMAType] = MMAType.F32_16x16x16_F16
+    vector_shapes: Optional[dict[IndexExpr, int]] = None
```

Comment:
Can we add description on the vector_shape to the comment on Hardwareconstraint? Is it vector shape of matmul per wave?

Reply:
Sure will add in a separate PR.

Reply:
It is the shape of the matmul per wave if there is an mma operation in the graph. Otherwise it is user-settable.